contracts/governance/GovernorAlphaZama.sol

```diff
@@ -0,0 +1,667 @@
+    ///         and for a vote to succeed.
+    /// @dev    It is set at 400,000, which is 4% of the total supply of the Comp token.
+    uint64 public constant QUORUM_VOTES = 400000e6;
+
+    /// @notice The delay before voting on a proposal may take place once proposed.
+    ///         It is 1 block.
+    uint256 public constant VOTING_DELAY = 1;
+
+    /// @notice The duration of voting on a proposal, in blocks
+    /// @dev    It is recommended to be set at 3 days in blocks
+    ///         (i.e 21,600 for 12-second blocks).
+    uint256 public immutable VOTING_PERIOD;
+
+    /// @notice Comp governance token.
+    IComp public immutable COMP;
+
+    /// @notice Compound Timelock.
+    ICompoundTimelock public immutable TIMELOCK;
+
+    /// @notice Constant for zero using TFHE.
+    /// @dev    Since it is expensive to compute 0, it is stored instead.
+    ///         However, is not possible to define it as constant due to TFHE constraints.
+    /* solhint-disable var-name-mixedcase*/
+    euint64 private _EUINT64_ZERO;
+
+    /// @notice Constant for PROPOSAL_THRESHOLD using TFHE.
+    /// @dev    Since it is expensive to compute 0, it is stored instead.
+    ///         However, is not possible to define it as constant due to TFHE constraints.
+    /* solhint-disable var-name-mixedcase*/
+    euint64 private _EUINT64_PROPOSAL_THRESHOLD;
+
+    /// @notice The total number of proposals made.
+    ///         It includes all proposals, including the ones that
+    ///         were rejected/canceled/defeated.
+    uint256 public proposalCount;
+
+    /// @notice The latest proposal for each proposer.
+    mapping(address proposer => uint256 proposalId) public latestProposalIds;
+
+    /// @notice Ballot receipt for an account for a proposal id.
+    mapping(uint256 proposalId => mapping(address => Receipt)) internal _accountReceiptForProposalId;
+
+    /// @notice The official record of all proposals that have been created.
+    mapping(uint256 proposalId => Proposal proposal) internal _proposals;
+
+    /// @notice Returns the proposal id associated with the request id from the Gateway.
+    /// @dev    This mapping is used for decryption.
+    mapping(uint256 requestId => uint256 proposalId) internal _requestIdToProposalId;
+
+    /**
+     * @param owner_            Owner address.
+     * @param timelock_         Timelock contract.
+     * @param comp_             Comp token.
+     * @param votingPeriod_     Voting period.
+     * @dev                     Do not use a small value in production such as 5 or 20 to avoid security issues
+     *                          unless for testing purpose. It should by at least a few days,.
+     *                          For instance, 3 days would have a votingPeriod = 21,600 blocks if 12s per block.
+     */
+    constructor(address owner_, address timelock_, address comp_, uint256 votingPeriod_) Ownable(owner_) {
+        TFHE.setFHEVM(FHEVMConfig.defaultConfig());
```

Comment:
You shouldn't use a config. The user will extend the contract and add the config in their constructor.